q2_dummy_types/_int_sequence.py

```diff
@@ -0,0 +1,70 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os.path
+
+import qiime.plugin
+
+from .plugin_setup import plugin
+
+IntSequence1 = qiime.plugin.SemanticType('IntSequence1')
+IntSequence2 = qiime.plugin.SemanticType('IntSequence2')
+
+plugin.register_semantic_type(IntSequence1)
+plugin.register_semantic_type(IntSequence2)
+
+
+###############################################################################
+#
+# int-sequence:
+#
+#     A sequence of integers stored in a single file. Since this is a sequence,
+#     the integers have an order and repetition of elements is allowed.
+#
+###############################################################################
+
+class IntSequenceFormat(qiime.plugin.FileFormat):
+    name = 'int-sequence'
+
+    @classmethod
+    def sniff(cls, filepath):
+        with open(filepath, 'r') as fh:
+            for line, _ in zip(fh, range(5)):
+                try:
+                    int(line.rstrip('\n'))
+                except (TypeError, ValueError):
+                    return False
+            return True
+
+int_sequence_data_layout = qiime.plugin.DataLayout('int-sequence', 1)
+int_sequence_data_layout.register_file('ints.txt', IntSequenceFormat)
+
+plugin.register_data_layout(int_sequence_data_layout)
+plugin.register_type_to_data_layout(IntSequence1, 'int-sequence', 1)
+plugin.register_type_to_data_layout(IntSequence2, 'int-sequence', 1)
+
+
```

Comment:
Could you comment these general sections explaining what the registrations are accomplishing? e.g. We are associating a reader / writer for a given data layout to a particular object view.